### PR TITLE
Ensure non-projection layers can have at most 2 slots + added Eye parameter to getSubImageView

### PR DESCRIPTION
**Pull request by cabanier**
This change ensures that non-projection layers are no longer tied to views. 
It also changes getSubImageView so it only works on projection layers

### CODE REVIEW

**Comment by cabanier** [comment]
Fixes #163

File: webxrlayers-1.bs [hunk 1]
```diff
@@ -907,10 +910,12 @@ To <dfn>determine the layout attribute</dfn> using an {{XRTextureType}} |texture
...
   1. If |layout| is {{XRLayerLayout/"mono"}}, return |layout| and abort these steps.
   1. If |layout| is {{XRLayerLayout/"default"}}, run the following steps:
+    1. If the size of [=list of views=] is <code>1</code>, return {{XRLayerLayout/"mono"}} and abort these steps.
     1. If |textureType| is {{"texture-array"}}, return |layout| and abort these steps.
+  1. If |layout| is {{XRLayerLayout/"default"}} or {{XRLayerLayout/"stereo"}}, run the following steps:
     1. If the user agent prefers {{XRLayerLayout/"stereo-left-right"}} layout, return {{XRLayerLayout/"stereo-left-right"}} and abort these steps.
     1. If the user agent prefers {{XRLayerLayout/"stereo-top-bottom"}} layout, return {{XRLayerLayout/"stereo-top-bottom"}} and abort these steps.
-  1. If |textureType| is {{"texture-array"}}, throw {{TypeError}} and abort these steps.
+  1. If |textureType| is {{"texture-array"}} and |layout| is {{XRLayerLayout/"stereo-left-right"}} or {{XRLayerLayout/"stereo-top-bottom"}}, throw {{TypeError}} and abort these steps.
```
Comment (toji) [inline webxrlayers-1.bs:918]: I don't particularly oppose throwing in this situation, but wanted to point out that a possibly exception is returning a texture array with a single layer. Sounds silly, but based on the call today we may find ourselves doing that in some scenarios anyway? Don't block this PR on it, though.

File: webxrlayers-1.bs [hunk 2]
```diff
@@ -859,8 +862,8 @@ interface XRWebGLBinding {
                                       optional XREquirectLayerInit init);
   XRCubeLayer createCubeLayer(optional XRCubeLayerInit init);
 
-  XRWebGLSubImage getSubImage(XRCompositionLayer layer, XRFrame frame);
-  XRWebGLSubImage getViewSubImage(XRCompositionLayer layer, XRView view);
+  XRWebGLSubImage getSubImage(XRCompositionLayer layer, XRFrame frame, optional XREye eye = "none");
+  XRWebGLSubImage getViewSubImage(XRProjectionLayer layer, XRView view);
```
Comment (toji) [inline webxrlayers-1.bs:866]: I really like the idea of explicitly limiting `getViewSubImage()` to `XRProjectionLayer`s! The only question it leaves me with is what happens if I pass a projection layer to `getSubImage()`? It's still an `XRCompositorLayer`, so the IDL won't reject it, but I don't think it's intended to work. We probably need a check in the algorithm below to reject it?

Comment (cabanier) [reply webxrlayers-1.bs:866]: I agree that it makes little sense that projection layers are allowed here.
I can't think of a good reason why an author would do that apart.

I will add the rejection of projection layers to the algorithm.

File: webxrlayers-1.bs (hunk 1 above)
Comment (cabanier) [reply webxrlayers-1.bs:918]: I agree. We don't limit mono layers to texture_2d either. 
I will make the change.

**Pull request merged**